catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +117,40 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        Serializable textFacetPropRaw = request.getPropertyValue(EXPERIMENTAL_FACET_FIELDS_KEY);
+
+        if(textFacetPropRaw != null && textFacetPropRaw instanceof FacetProperties) {
+            FacetProperties textFacetProp = (FacetProperties) textFacetPropRaw;
+            isFacetedQuery = true;
+
+            textFacetProp.getFacetFields().stream()
+                    .map(facet -> !facet.endsWith("_txt") ? facet + "_txt" : facet)
```

Comment:
❗️ You should use the metacard type and `DynamicSchemaResolver` to figure out the attribute type and its format.  You cannot assume that the given attribute to facet will be a String attribute.